src/tools/clippy/clippy_lints/src/wildcard_imports.rs

```diff
@@ -176,7 +176,8 @@ impl LateLintPass<'_> for WildcardImports {
                     format!("{import_source_snippet}::{imports_string}")
                 };
 
-                let (lint, message) = if let Res::Def(DefKind::Enum, _) = use_path.res {
+                // Glob imports always have a single resolution.
+                let (lint, message) = if let Res::Def(DefKind::Enum, _) = use_path.res[0] {
```

Comment:
This could be when already matching anyway. A little bit safer.
```suggestion
                let (lint, message) = if let Some(Res::Def(DefKind::Enum, _)) = use_path.res.get(0) {
```

Reply:
If glob import doesn't have a resolution (for its prefix) then something unexpected is happening, so it should be fine to leae this as indexing.